examples/source_examples/sql_server/connector.py

```diff
@@ -0,0 +1,230 @@
+# The function takes two parameters:
+# - configuration: dictionary contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary contains whatever state you have chosen to checkpoint during the prior sync
+# The state dictionary is empty for the first sync or for any full re-sync
+def update(configuration: dict, state: dict):
+    log.warning("Example: Source Examples - SQL Server")
+
+    # The yield statement returns a generator object.
+    # This generator will yield an upsert operation to the Fivetran connector.
+    # The op.upsert method is called with two arguments:
+    # - The first argument is the name of the table to upsert the data into, in this case, "employee_details".
+    # - The second argument is a dictionary containing the data to be upserted,
+
+    # This is not required. This is just for example illustration purposes.
+    setup_db(configuration)
+
+    # Fetch records from DB
+    query = "SELECT * FROM employee_details;"  # Replace with your table name
+
+    # Initialize variables
+    conn = None
+    cursor = None
+
+    try:
+        # Connect to your database instance instance.
+        conn = connect_to_database(configuration)
+        # Create a cursor from the connection
+        cursor = conn.cursor()
+        cursor.execute(query)
+
+        # Fetch and display results
+        print("Query executed successfully. Results:")
+        batch_size = 2 # Fetch only few records, process them and then fetch more, ensuring we don't load all the records in memory
+        while True:
+            # Fetch a batch of records.
+            rows = cursor.fetchmany(batch_size)
+            if not rows:
+                # Exit the loop when there are no more records.
+                break
+
+            for row in rows:
+                yield op.upsert(
+                    table="employee_details",
+                    data={
+                        "employee_id": row[0],  # Employee Id.
+                        "first_name": row[1],  # First Name.
+                        "last_name": row[2],  # Last Name.
+                        "hire_date": dt2str(row[3]),  # Hire Date.
+                        "salary": row[4]  # Salary.
+                    }
+                )
+    except pyodbc.Error as e:
+        print(f"Error executing query: {e}")
+    finally:
+        # Clean up and close the connection
+        if cursor is not None:
+            cursor.close()
+        if conn is not None:
+            conn.close()
+        print("Connection closed.")
+
+    # You need to checkpoint the state, to frequently save the progress, even with empty state checkpoint can be called.
+    yield op.checkpoint(state)
+
+
+# This creates the connector object that will use the update function defined in this connector.py file.
+# This example does not use the schema() function. If it did, it would need to be included in the connector object definition.
```

Comment:
Wrong comment

Reply:
This example does not use the schema() function. If it did, it would need to be included in the connector object definition.

Reply:
Please change it in the redshift example too

Reply:
will do it separately, I will scan other connectors too.

Reply:
for this one, fixed in this PR.